execution-plane/arcon/src/stream/operator/mod.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2020, KTH Royal Institute of Technology.
+// SPDX-License-Identifier: AGPL-3.0-only
+
+pub mod function;
+pub mod sink;
+pub mod window;
+
+use crate::data::{ArconElement, ArconEvent, ArconType, Epoch, Watermark};
+use crate::stream::channel::strategy::ChannelStrategy;
+use arcon_error::ArconResult;
+
+/// Defines the methods an `Operator` must implement
+pub trait Operator<IN, OUT>
+where
+    IN: ArconType,
+    OUT: ArconType,
+{
+    /// Determines how the `Operator` processes Elements
+    ///
+    /// The function takes an Element and a [ChannelStrategy] in order to pass on it
+    fn handle_element(&mut self, element: ArconElement<IN>, strategy: &mut ChannelStrategy<OUT>);
+    /// Determines how the `Operator` processes Watermarks
+    ///
+    /// The function either returns None or a Vec of ArconEvents (e.g., Window Computation)
+    fn handle_watermark(&mut self, watermark: Watermark) -> Option<Vec<ArconEvent<OUT>>>;
```

Comment:
No, not really. Just thought it was nicer to return a None rather than a Vec. 

But then again, the whole Operator trait is a bit confusing. handle_element is now taking a ChannelStrategy parameter rather than returning in most cases a single element using vec![elem]. I left the others as they were as they are not called as often..

TLDR: The Operator trait needs some more love.